src/ecdsa/serialized_signature.rs

```diff
@@ -0,0 +1,263 @@
+}
+
+impl fmt::Debug for SerializedSignature {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(self, f)
+    }
+}
+
+impl fmt::Display for SerializedSignature {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        for v in self {
+            write!(f, "{:02x}", v)?;
+        }
+        Ok(())
+    }
+}
+
+impl Default for SerializedSignature {
+    #[inline]
+    fn default() -> SerializedSignature {
+        SerializedSignature {
+            data: [0u8; MAX_LEN],
+            len: 0,
+        }
+    }
+}
+
+impl PartialEq for SerializedSignature {
+    #[inline]
+    fn eq(&self, other: &SerializedSignature) -> bool {
+        **self == **other
+    }
+}
+
+impl AsRef<[u8]> for SerializedSignature {
+    #[inline]
+    fn as_ref(&self) -> &[u8] {
+        &*self
+    }
+}
+
+impl ops::Deref for SerializedSignature {
+    type Target = [u8];
+
+    #[inline]
+    fn deref(&self) -> &[u8] {
+        &self.data[..self.len]
+    }
+}
+
+impl Eq for SerializedSignature {}
+
+impl IntoIterator for SerializedSignature {
+    type IntoIter = IntoIter;
+    type Item = u8;
+
+    #[inline]
+    fn into_iter(self) -> Self::IntoIter {
+        IntoIter::new(self)
+    }
+}
+
+impl<'a> IntoIterator for &'a SerializedSignature {
+    type IntoIter = core::slice::Iter<'a, u8>;
+    type Item = &'a u8;
+
+    #[inline]
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+impl SerializedSignature {
+    /// Get a pointer to the underlying data with the specified capacity.
+    #[inline]
+    pub(crate) fn get_data_mut_ptr(&mut self) -> *mut u8 {
+        self.data.as_mut_ptr()
+    }
+
+    /// Get the capacity of the underlying data buffer.
+    #[inline]
+    pub fn capacity(&self) -> usize {
+        self.data.len()
+    }
+
+    /// Get the len of the used data.
+    #[inline]
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    /// Set the length of the object.
+    #[inline]
+    pub(crate) fn set_len(&mut self, len: usize) {
+        self.len = len;
+    }
+
+    /// Convert the serialized signature into the Signature struct.
+    /// (This DER deserializes it)
+    #[inline]
+    pub fn to_signature(&self) -> Result<Signature, Error> {
+        Signature::from_der(self)
+    }
+
+    /// Create a SerializedSignature from a Signature.
+    /// (this DER serializes it)
+    #[inline]
+    pub fn from_signature(sig: &Signature) -> SerializedSignature {
+        sig.serialize_der()
+    }
+
+    /// Check if the space is zero.
+    #[inline]
+    pub fn is_empty(&self) -> bool { self.len() == 0 }
+}
+
+/// Separate mod to prevent outside code accidentally breaking invariants.
+mod into_iter {
+    use super::*;
+
+    /// Owned iterator over the bytes of [`SerializedSignature`]
+    ///
+    /// Created by [`IntoIterator::into_iter`] method.
+    // allowed because of https://github.com/rust-lang/rust/issues/98348
+    #[allow(missing_copy_implementations)]
```

Comment:
FYI this one just got resolved in the compiler. But I suppose we don't want warnings in MSRV?

Reply:
IMHO I'm fine with having warnings in MSRV. Just not errors.

Reply:
Is there any minimal version you'd like warning-free?

Reply:
IMHO no. I think we should CI-check with nightly (and accept that we'll get somewhat-frequent breakage) and that should be our critera. And I believe this is our current situation .. though maybe we check with `stable` which is also fine to me.

But I want to emphasize that I don't feel very strongly at all about this, and if you wanted a policy of "clippy runs clean on some version of rustc" I'd be happy to do that...though we should promote this to a discussion rather than commenting on a closed PR :)

Reply:
Sure https://github.com/rust-bitcoin/rust-bitcoin/discussions/1456